src/libsystemd-network/sd-dhcp-lease.c

```diff
@@ -678,6 +705,18 @@ int sd_dhcp_lease_save(sd_dhcp_lease *lease, const char *lease_file) {
         if (r >= 0)
                 serialize_dhcp_routes(f, "ROUTES", routes, r);
 
+        r = sd_dhcp_lease_get_vendor_specific(lease, &data, &data_len);
```

Comment:
The function, as all other sd_dhcp_lease_get_*() functions, is added
as part of "public" API of libsystemd-network; it isn't used anywhere
else at the moment (like sd_dhcp_lease_get_server_identifier() and
sd_dhcp_lease_get_root_path()) but IMHO it shoud be made available.

Beside this, NetworkManager is reusing part of libsystemd-network code
to provide an optional internal DHCP client and this function would be
useful there.